File: src/main/java/dev/boooiil/historia/core/classes/enums/FileMap.java
package dev.boooiil.historia.core.classes.enums;

public class FileMap {
    /**
     * This enum represents the keys for the various resource files used in the
     * plugin.
     */
    public enum ResourceKeys {

        /** armor.yml */
        ARMOR("armor.yml"),
        /** proficiency.yml */
        PROFICIENCY("proficiency.yml"),
        /** config.yml */
        CONFIG("config.yml"),
        /** plugin.yml */
        PLUGIN("plugin.yml"),
        /** weapons.yml */
        WEAPONS("weapons.yml"),
        /** tools.yml */
        TOOLS("tools.yml");

        private final String key;

        ResourceKeys(String key) {

            this.key = key;

        }

        public String getKey() {

            return this.key;

        }

    }

}

File: src/main/java/dev/boooiil/historia/core/expiry/configuration/FileMap.kt
package dev.boooiil.historia.core.expiry.configuration

class FileMap {
    /**
     * This enum represents the keys for the various resource files used in the
     * plugin.
     */
    enum class ResourceKeys(val key: String) {
        /** config.yml  */
        CONFIG("config.yml"),
        ITEMS("items.yml");
    }
}
